File: CURRENT/java/src/org/mariuszgromada/math/mxparser/parsertokens/BinaryRelation.java
/*
 * @(#)BinaryRelation.java        4.1.0    2017-05-13
 *
 * You may use this software under the condition of "Simplified BSD License"
 *
 * Copyright 2010-2017 MARIUSZ GROMADA. All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without modification, are
 * permitted provided that the following conditions are met:
 *
 *    1. Redistributions of source code must retain the above copyright notice, this list of
 *       conditions and the following disclaimer.
 *
 *    2. Redistributions in binary form must reproduce the above copyright notice, this list
 *       of conditions and the following disclaimer in the documentation and/or other materials
 *       provided with the distribution.
 *
 * THIS SOFTWARE IS PROVIDED BY <MARIUSZ GROMADA> ``AS IS'' AND ANY EXPRESS OR IMPLIED
 * WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
 * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL <COPYRIGHT HOLDER> OR
 * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
 * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
 * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
 * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
 * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
 * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 *
 * The views and conclusions contained in the software and documentation are those of the
 * authors and should not be interpreted as representing official policies, either expressed
 * or implied, of MARIUSZ GROMADA.
 *
 * If you have any questions/bugs feel free to contact:
 *
 *     Mariusz Gromada
 *     mariuszgromada.org@gmail.com
 *     http://mathparser.org
 *     http://mathspace.pl
 *     http://janetsudoku.mariuszgromada.org
 *     http://github.com/mariuszgromada/MathParser.org-mXparser
 *     http://mariuszgromada.github.io/MathParser.org-mXparser
 *     http://mxparser.sourceforge.net
 *     http://bitbucket.org/mariuszgromada/mxparser
 *     http://mxparser.codeplex.com
 *     http://github.com/mariuszgromada/Janet-Sudoku
 *     http://janetsudoku.codeplex.com
 *     http://sourceforge.net/projects/janetsudoku
 *     http://bitbucket.org/mariuszgromada/janet-sudoku
 *     http://github.com/mariuszgromada/MathParser.org-mXparser
 *     http://scalarmath.org/
 *     https://play.google.com/store/apps/details?id=org.mathparser.scalar.lite
 *     https://play.google.com/store/apps/details?id=org.mathparser.scalar.pro
 *
 *                              Asked if he believes in one God, a mathematician answered:
 *                              "Yes, up to isomorphism."
 */
package org.mariuszgromada.math.mxparser.parsertokens;

import org.mariuszgromada.math.mxparser.mXparser;
/**
 * Binary Relations - mXparser tokens definition.
 *
 * @author         <b>Mariusz Gromada</b><br>
 *                 <a href="mailto:mariuszgromada.org@gmail.com">mariuszgromada.org@gmail.com</a><br>
 *                 <a href="http://mathspace.pl" target="_blank">MathSpace.pl</a><br>
 *                 <a href="http://mathparser.org" target="_blank">MathParser.org - mXparser project page</a><br>
 *                 <a href="http://github.com/mariuszgromada/MathParser.org-mXparser" target="_blank">mXparser on GitHub</a><br>
 *                 <a href="http://mxparser.sourceforge.net" target="_blank">mXparser on SourceForge</a><br>
 *                 <a href="http://bitbucket.org/mariuszgromada/mxparser" target="_blank">mXparser on Bitbucket</a><br>
 *                 <a href="http://mxparser.codeplex.com" target="_blank">mXparser on CodePlex</a><br>
 *                 <a href="http://janetsudoku.mariuszgromada.org" target="_blank">Janet Sudoku - project web page</a><br>
 *                 <a href="http://github.com/mariuszgromada/Janet-Sudoku" target="_blank">Janet Sudoku on GitHub</a><br>
 *                 <a href="http://janetsudoku.codeplex.com" target="_blank">Janet Sudoku on CodePlex</a><br>
 *                 <a href="http://sourceforge.net/projects/janetsudoku" target="_blank">Janet Sudoku on SourceForge</a><br>
 *                 <a href="http://bitbucket.org/mariuszgromada/janet-sudoku" target="_blank">Janet Sudoku on BitBucket</a><br>
 *                 <a href="https://play.google.com/store/apps/details?id=org.mathparser.scalar.lite" target="_blank">Scalar Free</a><br>
 *                 <a href="https://play.google.com/store/apps/details?id=org.mathparser.scalar.pro" target="_blank">Scalar Pro</a><br>
 *                 <a href="http://scalarmath.org/" target="_blank">ScalarMath.org</a><br>
 *
 * @version        4.1.0
 */
public final class BinaryRelation {
	/*
	 * BinaryRelation - token type id.
	 */
	public static final int TYPE_ID 				= 3;
	public static final String TYPE_DESC			= "Binary Relation";
	/*
	 * BinaryRelation - tokens id.
	 */
	public static final int EQ_ID 					= 1;
	public static final int NEQ_ID 					= 2;
	public static final int LT_ID 					= 3;
	public static final int GT_ID 					= 4;
	public static final int LEQ_ID 					= 5;
	public static final int GEQ_ID 					= 6;
	/*
	 * BinaryRelation - tokens key words.
	 */
	public static final String EQ_STR 				= "=";
	public static final String EQ1_STR 				= "==";
	public static final String NEQ_STR 				= "<>";
	public static final String NEQ1_STR 			= "~=";
	public static final String NEQ2_STR 			= "!=";
	public static final String LT_STR 				= "<";
	public static final String GT_STR 				= ">";
	public static final String LEQ_STR 				= "<=";
	public static final String GEQ_STR 				= ">=";
	/*
	 * BinaryRelation - syntax.
	 */
	public static final String EQ_SYN 				= "a = b";
	public static final String EQ1_SYN 				= "a == b";
	public static final String NEQ_SYN 				= "a <> b";
	public static final String NEQ1_SYN 			= "a ~= b";
	public static final String NEQ2_SYN 			= "a != b";
	public static final String LT_SYN 				= "a < b";
	public static final String GT_SYN 				= "a > b";
	public static final String LEQ_SYN 				= "a <= b";
	public static final String GEQ_SYN 				= "a >= b";
	/*
	 * BinaryRelation - tokens description.
	 */
	public static final String EQ_DESC 				= "Equality";
	public static final String NEQ_DESC				= "Inequation";
	public static final String LT_DESC 				= "Lower than";
	public static final String GT_DESC 				= "Greater than";
	public static final String LEQ_DESC 			= "Lower or equal";
	public static final String GEQ_DESC 			= "Greater or equal";
	/*
	 * BinaryRelation - since.
	 */
	public static final String EQ_SINCE				= mXparser.NAMEv10;
	public static final String NEQ_SINCE			= mXparser.NAMEv10;
	public static final String LT_SINCE 			= mXparser.NAMEv10;
	public static final String GT_SINCE				= mXparser.NAMEv10;
	public static final String LEQ_SINCE 			= mXparser.NAMEv10;
	public static final String GEQ_SINCE 			= mXparser.NAMEv10;
}

File: CURRENT/typescript/org/mariuszgromada/math/mxparser/parsertokens/BinaryRelation.ts
import { mXparserConstants } from '../mXparserConstants';

/**
 * Binary Relations - mXparser tokens definition.
 * 
 * @author         <b>Mariusz Gromada</b><br>
 * <a href="mailto:mariuszgromada.org@gmail.com">mariuszgromada.org@gmail.com</a><br>
 * <a href="http://github.com/mariuszgromada/MathParser.org-mXparser" target="_blank">mXparser on GitHub</a><br>
 * 
 * @version        4.1.0
 * @class
 */
export class BinaryRelation {
    public static TYPE_ID: number = 3;

    public static TYPE_DESC: string = "Binary Relation";

    public static EQ_ID: number = 1;

    public static NEQ_ID: number = 2;

    public static LT_ID: number = 3;

    public static GT_ID: number = 4;

    public static LEQ_ID: number = 5;

    public static GEQ_ID: number = 6;

    public static EQ_STR: string = "=";

    public static EQ1_STR: string = "==";

    public static NEQ_STR: string = "<>";

    public static NEQ1_STR: string = "~=";

    public static NEQ2_STR: string = "!=";

    public static LT_STR: string = "<";

    public static GT_STR: string = ">";

    public static LEQ_STR: string = "<=";

    public static GEQ_STR: string = ">=";

    public static EQ_SYN: string = "a = b";

    public static EQ1_SYN: string = "a == b";

    public static NEQ_SYN: string = "a <> b";

    public static NEQ1_SYN: string = "a ~= b";

    public static NEQ2_SYN: string = "a != b";

    public static LT_SYN: string = "a < b";

    public static GT_SYN: string = "a > b";

    public static LEQ_SYN: string = "a <= b";

    public static GEQ_SYN: string = "a >= b";

    public static EQ_DESC: string = "Equality";

    public static NEQ_DESC: string = "Inequation";

    public static LT_DESC: string = "Lower than";

    public static GT_DESC: string = "Greater than";

    public static LEQ_DESC: string = "Lower or equal";

    public static GEQ_DESC: string = "Greater or equal";

    public static EQ_SINCE: string; public static EQ_SINCE_$LI$(): string { if (BinaryRelation.EQ_SINCE == null) { BinaryRelation.EQ_SINCE = mXparserConstants.NAMEv10; }  return BinaryRelation.EQ_SINCE; }

    public static NEQ_SINCE: string; public static NEQ_SINCE_$LI$(): string { if (BinaryRelation.NEQ_SINCE == null) { BinaryRelation.NEQ_SINCE = mXparserConstants.NAMEv10; }  return BinaryRelation.NEQ_SINCE; }

    public static LT_SINCE: string; public static LT_SINCE_$LI$(): string { if (BinaryRelation.LT_SINCE == null) { BinaryRelation.LT_SINCE = mXparserConstants.NAMEv10; }  return BinaryRelation.LT_SINCE; }

    public static GT_SINCE: string; public static GT_SINCE_$LI$(): string { if (BinaryRelation.GT_SINCE == null) { BinaryRelation.GT_SINCE = mXparserConstants.NAMEv10; }  return BinaryRelation.GT_SINCE; }

    public static LEQ_SINCE: string; public static LEQ_SINCE_$LI$(): string { if (BinaryRelation.LEQ_SINCE == null) { BinaryRelation.LEQ_SINCE = mXparserConstants.NAMEv10; }  return BinaryRelation.LEQ_SINCE; }

    public static GEQ_SINCE: string; public static GEQ_SINCE_$LI$(): string { if (BinaryRelation.GEQ_SINCE == null) { BinaryRelation.GEQ_SINCE = mXparserConstants.NAMEv10; }  return BinaryRelation.GEQ_SINCE; }
}
BinaryRelation["__class"] = "org.mariuszgromada.math.mxparser.parsertokens.BinaryRelation";